app/assets/javascripts/components/AccountMenu.tsx

```diff
@@ -0,0 +1,984 @@
+      : hasPasscode
+        ? STRING_LOCAL_ENC_ENABLED
+        : STRING_ENC_NOT_ENABLED;
+
+    setEncryptionStatusString(newEncryptionStatusString);
+    setIsEncryptionEnabled(encryptionEnabled);
+    setIsBackupEncrypted(encryptionEnabled);
+  }, [application]);
+
+  const handleAddPassCode = () => {
+    setShowPasscodeForm(true);
+
+    // At this moment the passcode input is not available, therefore the ref
+    // is null. Therefore we call `focus()` after timeout.
+    focusWithTimeout(passcodeInputRef);
+  };
+
+  const submitPasscodeForm = async (event: TargetedEvent<HTMLFormElement> | TargetedMouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+
+    if (passcode !== passcodeConfirmation) {
+      await alertDialog({
+        text: STRING_NON_MATCHING_PASSCODES
+      });
+      passcodeInputRef.current.focus();
+      return;
+    }
+
+    await preventRefreshing(
+      STRING_CONFIRM_APP_QUIT_DURING_PASSCODE_CHANGE,
+      async () => {
+        const successful = application.hasPasscode()
+          ? await application.changePasscode(passcode as string)
+          : await application.addPasscode(passcode as string);
+
+        if (!successful) {
+          passcodeInputRef.current.focus();
+        }
+      }
+    );
+
+    setPasscode(undefined);
+    setPasscodeConfirmation(undefined);
+    setShowPasscodeForm(false);
+
+    setProtectionsDisabledUntil(getProtectionsDisabledUntil());
+
+    refreshEncryptionStatus();
+  };
+
+  const handlePasscodeChange = (event: TargetedEvent<HTMLInputElement>) => {
+    const { value } = event.target as HTMLInputElement;
+    setPasscode(value);
+  };
+
+  const handleConfirmPasscodeChange = (event: TargetedEvent<HTMLInputElement>) => {
+    const { value } = event.target as HTMLInputElement;
+    setPasscodeConfirmation(value);
+  };
+
+  const selectAutoLockInterval = async (interval: number) => {
+    if (!(await application.authorizeAutolockIntervalChange())) {
+      return;
+    }
+    await application.getAutolockService().setAutoLockInterval(interval);
+    reloadAutoLockInterval();
+  };
+
+  const disableBetaWarning = () => {
+    console.log('disableBetaWarning');
+  };
+
+  const hidePasswordForm = () => {
+    setShowLogin(false);
+    setShowRegister(false);
+    setPassword('');
+    setPasswordConfirmation(undefined);
+  };
+
+  const signOut = () => {
+    appState.accountMenu.setSigningOut(true);
+  };
+
+  const changePasscodePressed = () => {
+    handleAddPassCode();
+  };
+
+  const removePasscodePressed = async () => {
+    await preventRefreshing(
+      STRING_CONFIRM_APP_QUIT_DURING_PASSCODE_REMOVAL,
+      async () => {
+        if (await application.removePasscode()) {
+          await application
+            .getAutolockService()
+            .deleteAutolockPreference();
+          await reloadAutoLockInterval();
+          refreshEncryptionStatus();
+        }
+      }
+    );
+
+    setProtectionsDisabledUntil(getProtectionsDisabledUntil());
```

Comment:
Once we have the event observer for ProtectionSessionExpiryDateChanged, I think we won't need to manually set the protectionsDisabledUntil state

Reply:
Correct, I removed them